src/components/Calculator.js

```diff
@@ -1,58 +1,47 @@
-          <button className="btn" onClick={this.handleEvent} type="button">2</button>
-          <button className="btn" onClick={this.handleEvent} type="button">3</button>
-          <button className="btn calc-zero" onClick={this.handleEvent} type="button">0</button>
-          <button className="btn btn-point" onClick={this.handleEvent} type="button">.</button>
-        </div>
-        <div className="calc-operators">
-          <button className="btn-operators" onClick={this.handleEvent} type="button">÷</button>
-          <button className="btn-operators" onClick={this.handleEvent} type="button">x</button>
-          <button className="btn-operators" onClick={this.handleEvent} type="button">-</button>
-          <button className="btn-operators" onClick={this.handleEvent} type="button">+</button>
-          <button className="btn-operators" onClick={this.handleEvent} type="button">=</button>
-        </div>
+  const handleEvent = (e) => {
+    setState(calculate(previousState, e.target.innerText));
+  };
 
+  return (
+    <div className="calculator">
+      <div className="calc-display">
+        {previousState.total}
+        {previousState.operation}
+        {previousState.next}
+      </div>
+      <div className="calc-main">
+        <button className="btn" onClick={handleEvent} type="button">AC</button>
+        <button className="btn" onClick={handleEvent} type="button">+/-</button>
+        <button className="btn" onClick={handleEvent} type="button">%</button>
+        <button className="btn" onClick={handleEvent} type="button">7</button>
+        <button className="btn" onClick={handleEvent} type="button">8</button>
+        <button className="btn" onClick={handleEvent} type="button">9</button>
+        <button className="btn" onClick={handleEvent} type="button">4</button>
+        <button className="btn" onClick={handleEvent} type="button">5</button>
+        <button className="btn" onClick={handleEvent} type="button">6</button>
+        <button className="btn" onClick={handleEvent} type="button">1</button>
+        <button className="btn" onClick={handleEvent} type="button">2</button>
+        <button className="btn" onClick={handleEvent} type="button">3</button>
+        <button className="btn calc-zero" onClick={handleEvent} type="button">0</button>
+        <button className="btn btn-point" onClick={handleEvent} type="button">.</button>
       </div>
+      <div className="calc-operators">
+        <button className="btn-operators" onClick={handleEvent} type="button">÷</button>
+        <button className="btn-operators" onClick={handleEvent} type="button">x</button>
+        <button className="btn-operators" onClick={handleEvent} type="button">-</button>
+        <button className="btn-operators" onClick={handleEvent} type="button">+</button>
+        <button className="btn-operators" onClick={handleEvent} type="button">=</button>
+      </div>
```

Comment:
- [ ] (Optional) Kindly consider storing the button in an array in the order they appear and in this div, map through the array and render them dynamically to the `ReactDom`. Your code could look like this:

```
const buttonNames = ['AC', '+/-', '%', '\u00F7', '7', '8', '9', '\u00D7',
    '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
```

Then in your return function, you could do this:

```
<div className="button-grid">
            {buttonNames.map((button) => (
              <button
                key={button}
                className={generateClassName(button)}
                onClick={buttonClicked}
                type="button"
              >
                {button}
            </button>
            ))}
</div>
```

Reply:
Thanks, I created an issue to improve the code on this way